src/components/ProtectedRoute.test.jsx

```diff
@@ -0,0 +1,40 @@
+import React from 'react';
+import { render, screen } from '@testing-library/react';
+import { MemoryRouter, Route } from 'react-router-dom';
+import { clearFirestoreData, initializeTestApp } from '../lib/testing';
+import ScholarshipForm from '../pages/ScholarshipForm';
+
+window.MutationObserver = require('mutation-observer');
+
+function renderAtRoute(route) {
+  return render(
+    <MemoryRouter initialEntries={[route]}>
+      <Route path="/scholarships/new" component={ScholarshipForm} />
+    </MemoryRouter>
+  );
+}
+
+const app = initializeTestApp({
+  projectId: 'protected-route-test',
+  apiKey: 'fake-api-key',
+});
+
+beforeAll(() => clearFirestoreData(app.options));
+afterAll(() => app.delete());
+
+test('athenticated user can accesss page', () => {
```

Comment:
Does this test work?

Technically there are three test cases:
1. Initially the **CircularProgress** is displayed when `user` is undefined.
2. "You must log in first to access this page" when the `user` is not signed in.
3. The passed in component is displayed when `user` is signed in.

I'm not sure exactly how to mock the user but something you can manually set `firebase.auth().currentUser`.

Maybe you can even test the `onAuthStateChanged` thing using [firebase.auth().updateCurrentUser()](https://firebase.google.com/docs/reference/js/auth.auth.md#authupdatecurrentuser) and collapse the first test case with the other two.

Reply:
*authenticated